src/presenters/abstract-indexable-tag-presenter.php

```diff
@@ -36,12 +36,15 @@ abstract class Abstract_Indexable_Tag_Presenter extends Abstract_Indexable_Prese
 	 * @throws \InvalidArgumentException When a subclass does not define a key property. This should appear during development.
 	 */
 	public function present() {
+		$value = $this->get();
+
 		if ( $this->key === 'NO KEY PROVIDED' ) {
-			throw new \InvalidArgumentException( \get_class( $this ) . ' is an Abstract_Indexable_Presenter but does not override the key property.' );
+			/**
+			 * Required for backwards compatability with add-ons in which we override this class and define the key in the tag_format.
+			 */
+			return \sprintf( $this->tag_format, $this->escape_value( $value ) );
```

Comment:
@Xyfi I think you need to check `if ( \is_string( $value ) && $value !== '' ) {` here too, before returning the `/sprintf `

Reply:
I'd simply remove the `if ( $this->key === 'NO KEY PROVIDED' ) {` block entirely in this case.

Then in the `Meta` class make sure the `create_json_field` does get this clause and in that case returns `null`